impeller/renderer/backend/vulkan/device_buffer_vk.cc

```diff
@@ -46,6 +46,7 @@ bool DeviceBufferVK::OnCopyHostBuffer(const uint8_t* source,
   if (source) {
     ::memmove(dest + offset, source + source_range.offset, source_range.length);
   }
+  ::vmaFlushAllocation(allocator_, allocation_, offset, source_range.length);
```

Comment:
`#ifdef FML_OS_ANDROID`?

Reply:
are you sure? we can't actually test all the devices this will run on, and this will always be safe to call...

Reply:
I think we should probably just not request coherent memory if we don't need it and add the explict flushes

Reply:
Ok.

It looks like we should be querying `vmaGetMemoryProperties` to see if the https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkPhysicalDeviceMemoryProperties.html has a VkMemoryType that supports host coherent.